sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/integrationtest/BeamSqlBuiltinFunctionsIntegrationTestBase.java

```diff
@@ -180,13 +264,27 @@ public void buildRunAndCheck() {
 
         PCollection<Row> output =
             inputCollection.apply(testCase.toString(), SqlTransform.query(sql));
-
-        PAssert.that(output)
-            .containsInAnyOrder(TestUtils.RowsBuilder.of(schema).addRows(expectedValue).getRows());
+        if (testCase.isCompareNonExactNumber()) {
+          PAssert.that(output).satisfies(matchesScalar((double) expectedValue, PRECISION));
+        } else {
+          PAssert.that(output)
+              .containsInAnyOrder(
+                  TestUtils.RowsBuilder.of(schema).addRows(expectedValue).getRows());
+        }
       }
 
       inputCollection.getPipeline().run();
     }
+
+    private TypeName getTypeName(Object expectedValue) {
+      // Because of erasure, we can only automatically infer non-parameterized types
+      TypeName resultTypeName = JAVA_CLASS_TO_TYPENAME.get(expectedValue.getClass());
+      checkArgument(
+          resultTypeName != null,
+          "Could not infer a Beam type for %s."
+              + " Parameterized types must be provided explicitly.");
```

Comment:
Add `expectedValue` to the parameter of `checkArgument`.
Update the message to something like `"You're calling 'addExpr(String expr, Object expected)' for unknown type of the expected object '%s'. Please use 'addExpr(String expr, Object expected, FieldType type)' overload instead and provide the type of the expected object"`. Otherwise it's hard to understand what should be done in this case.